### PR TITLE
Metadata Timeline

### LINKED ISSUE #3856: Ability to debug recordings in the UI
**Describe what you are trying to accomplish and why in non technical terms**
I am currently working through missed detections and false positives, the only way to re-test changes to my settings is to wait for them to happen again or try to re-create them (IE go out walk in front of a camera etc)

**Describe the solution you'd like**
In the debug view have the option to select a previous recording and run it through the debug process to see how detection would behave on the recording after changing the settings.

**Describe alternatives you've considered**
Just doing what I am doing now.. go out walk in front of the camera, move a car, wait for more events. The problem with waiting is that you have to go back and review the recording to see if something was missed.

**Additional context**
I think this would be a VERY powerful diagnostic / setup tool since the project is focused on object detection, it would drastically cut down on tuning time.

### PR DESCRIPTION
**Pull request by NickM-27**
This PR accomplishes a few things as well as lays the groundwork for other features
1. New timeline table that keeps track of key moments in a tracked objects lifecycle and maintains metadata related to those moments
2. Adds UI support for viewing tracked objects and those key moments
3. Adds an api for grabbing a frame from a specific point in recording #5564
4. Lays the groundwork for audio and api events to be stored in the timeline

Closes #3856

https://user-images.githubusercontent.com/14866235/233727243-4d2aaef0-be39-4ed2-9ad9-f49d35973a49.mov

### CODE REVIEW

File: web/src/routes/Events.jsx [hunk 1]
```diff
@@ -365,7 +379,7 @@ export default function Events({ path, ...props }) {
               download
             />
           )}
-          {(downloadEvent.end_time && downloadEvent.has_snapshot && !downloadEvent.plus_id) && (
+          {downloadEvent.end_time && downloadEvent.has_snapshot && !downloadEvent.plus_id && (
```
Comment (blakeblackshear) [inline web/src/routes/Events.jsx:382]: looks like our vscode settings are fighting

File: web/src/routes/Events.jsx [hunk 2]
```diff
@@ -573,20 +587,52 @@ export default function Events({ path, ...props }) {
...
+                                    <div
+                                      className="absolute border-4 border-red-600"
+                                      style={{left: `${Math.round(
+                                        eventOverlay.data.box[0] * 100
+                                      )}%`, top: `${Math.round(eventOverlay.data.box[1] * 100)}%`, right: `${Math.round(
+                                        (1 - eventOverlay.data.box[2]) * 100
+                                      )}%`, bottom: `${Math.round((1 - eventOverlay.data.box[3]) * 100)}%`}}
+                                    >
+                                      {eventOverlay.class_type == 'entered_zone' ? (
+                                        <div className="absolute w-2 h-2 bg-red-600 left-[50%] bottom-0" />
```
Comment (blakeblackshear) [inline web/src/routes/Events.jsx:629]: lets make the box for the bottom center green or yellow instead so its more obvious

File: web/src/routes/Events.jsx (hunk 1 above)
Comment (NickM-27) [reply web/src/routes/Events.jsx]: is there a specific setting or place you have yours setup from? I think I am just using whatever vscode decided, if there's a template hopefully I can import that

**Review by blakeblackshear — APPROVED** [review]

**Comment by NickM-27** [comment]
No, but it has already been added back in for 0.15

**Comment by mbay0r** [comment]
> No, but it has already been added back in for 0.15

Thanks for the info! Are there already approximate dates for version 0.15? Thanks

**Comment by NickM-27** [comment]
we never have approximate dates, the releases are ready whenever they are ready. If you are eager to try it then you can use a dev build with the knowledge that there may be bugs

**Comment by mbay0r** [comment]
> we never have approximate dates, the releases are ready whenever they are ready. If you are eager to try it then you can use a dev build with the knowledge that there may be bugs

Thanks a lot, will give it a try!

**Pull request merged**